vendor/k8s.io/kubernetes/pkg/kubectl/genericclioptions/printers/interface.go

```diff
@@ -30,8 +30,15 @@ func (fn ResourcePrinterFunc) PrintObj(obj runtime.Object, w io.Writer) error {
 	return fn(obj, w)
 }
 
+// PrintObj implements ResourcePrinter
+func (fn ResourcePrinterFunc) PrintfObj(format string, obj runtime.Object, w io.Writer) error {
+	return fn(obj, w)
+}
+
 // ResourcePrinter is an interface that knows how to print runtime objects.
 type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
 	PrintObj(runtime.Object, io.Writer) error
+	// Print receives a runtime object, formats it according to passed format and prints it to a writer.
+	PrintfObj(string, runtime.Object, io.Writer) error
```

Comment:
I'm inclined to change the interface to this always, so that it's simple.

Reply:
Having this also allows us to get rid of that `Complete` method on `PrintFlags` since it's not needed anymore.